### PR TITLE
Deploy `PurchaseTesterSwiftUI` to TestFlight

### PR DESCRIPTION
**Pull request by NachoSoto**
Fixes [CSDK-510].

IMG_495339BA4A2F-1

Depends on #1999.

Also set up CI as per [the docs](https://docs.fastlane.tools/best-practices/continuous-integration/#storing-a-manually-verified-session-using-spaceauth):
Added these 2 environment variables:
- `SPACESHIP_ONLY_ALLOW_INTERACTIVE_2FA`
- `FASTLANE_SESSION`

[CSDK-510]: https://revenuecats.atlassian.net/browse/CSDK-510?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by NachoSoto** [comment]
This is ready 🎉

**Comment by NachoSoto** [comment]
I'm going to submit a catalyst version too

File: fastlane/Fastfile [hunk 1]
```diff
@@ -465,6 +465,28 @@ platform :ios do
     end
   end
 
+  desc "Build and deploy PurchaseTesterSwiftUI"
+  lane :deploy_purchase_tester do
+    match(verbose: true, readonly: true)
+    increment_build_number(
+      build_number: latest_testflight_build_number + 1,
```
Comment (tonidero) [inline fastlane/Fastfile:472]: I'm using the CircleCI build number here, but then, this is probably fine as well 👍

File: Tests/TestingApps/PurchaseTesterSwiftUI/fastlane/Fastfile [hunk 2]
```diff
@@ -0,0 +1 @@
+import("../../../../fastlane/Fastfile")
```
Comment (tonidero) [inline Tests/TestingApps/PurchaseTesterSwiftUI/fastlane/Fastfile:1]: With this I guess the `PurchaseTesterSwiftUI` project will have access to all the project fastfile's functions... There are probably many that shouldn't be called from here, but then again, I think it should be ok.

**Review by tonidero — APPROVED** [review]
Looks great!

File: Tests/TestingApps/PurchaseTesterSwiftUI/fastlane/Fastfile (hunk 2 above)
Comment (NachoSoto) [reply Tests/TestingApps/PurchaseTesterSwiftUI/fastlane/Fastfile:1]: Yeah it's fine. It's the same approach we use in the installation tests.

File: Tests/TestingApps/PurchaseTesterSwiftUI/AppIcon.xcassets/Contents.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (aboedo) [inline Tests/TestingApps/PurchaseTesterSwiftUI/AppIcon.xcassets/Contents.json:1]: the icon is glorious 🎨

File: fastlane/Fastfile (hunk 1 above)
Comment (aboedo) [reply fastlane/Fastfile:472]: I think the latest Xcode versions were supposed to auto-bump build numbers? 
But then again, that might be only if you use Xcode, not if you go through command line

**Review by aboedo — APPROVED** [review]

File: fastlane/Fastfile (hunk 1 above)
Comment (NachoSoto) [reply fastlane/Fastfile:472]: Yeah and I use that for my own app now, but it wasn't working without bumping the build number manually like this. So they probably built that feature into the UI only (like when Swift shipped 😂)

**Pull request merged**